File: webApp/src/app/common/order.ts
/**
 * 订单实体
 */
export class Order {
  /** id */
  id: number;

  /** 订单时间 */
  orderTime: string;

  /** 支付时间 */
  payTime: string;

  /** 订单状态 */
  status: string;

  /** 用户编号 */
  userId: number;

  /** 总价 */
  price: string;

  constructor(data?: { id?: number, orderTime?: string, payTime?: string, status?: string, userId?: number, price?: string }) {
    if (data) {
      this.id = data.id;
      this.orderTime = data.orderTime;
      this.payTime = data.payTime;
      this.status = data.status;
      this.userId = data.userId;
      this.price = data.price;
    }
  }
}

File: bus-ticket/src/main/java/xiaoqiangZzz/busticket/Entity/Order.java
package xiaoqiangZzz.busticket.Entity;


import org.springframework.data.annotation.CreatedDate;

import javax.persistence.*;

@Entity
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @CreatedDate
    private Long orderTime;

    private Long payTime;

    private int status;

    @ManyToOne
    @JoinColumn(nullable = false)
    private User user;

    private float price;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getOrderTime() {
        return orderTime;
    }

    public void setOrderTime(Long orderTime) {
        this.orderTime = orderTime;
    }

    public Long getPayTime() {
        return payTime;
    }

    public void setPayTime(Long payTime) {
        this.payTime = payTime;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }
}
